exporters/deploytime/app.py

```diff
@@ -50,54 +54,75 @@ def image_sha(img_url):
-        replicas = replicationcontrollers.items
-
-        # Process ReplicaSets from apps/v1 api version for Deployments
-        v1_replicationsets = dyn_client.resources.get(api_version='apps/v1', kind='ReplicaSet')
-        replicationsets = v1_replicationsets.get(namespace=namespace,
-                                                 label_selector=pelorus.get_app_label())
-        replicas = replicas + replicationsets.items
-
-        # Process ReplicaSets from extentions/v1beta1 api version for Deployments
-        v1_replicationsets = dyn_client.resources.get(api_version='extensions/v1beta1', kind='ReplicaSet')
-        replicationsets = v1_replicationsets.get(namespace=namespace,
-                                                 label_selector=pelorus.get_app_label())
-        replicas = replicas + replicationsets.items
-
-        for rc in replicas:
-            images = [image_sha(c.image) for c in rc.spec.template.spec.containers]
-            print("IMAGE" + str(images))
-
-            # Since a commit will be built into a particular image and there could be multiple
-            # containers (images) per pod, we will push one metric per image/container in the
-            # pod template
-            for i in images:
-                if i is not None:
-                    metric = DeployTimeMetric(rc.metadata.name, namespace)
-                    metric.labels = rc.metadata.labels
-                    metric.deploy_time = rc.metadata.creationTimestamp
-                    metric.image_sha = i
-                    metrics.append(metric)
+        logging.info("Watching namespaces %s", namespaces)
+
+    pods = v1_pods.get(label_selector=pelorus.get_app_label(), field_selector='status.phase=Running').items
+
+    replicas_dict = {}
+    # Process ReplicationControllers for DeploymentConfigs
+    replicas_dict = get_replicas('v1', 'ReplicationController')
+
+    # Process ReplicaSets from apps/v1 api version for Deployments
+    replicas_dict.update(get_replicas('apps/v1', 'ReplicaSet'))
+
+    # Process ReplicaSets from extentions/v1beta1 api version for Deployments
+    replicas_dict.update(get_replicas('extensions/v1beta1', 'ReplicaSet'))
+
+    for pod in pods:
+        if not namespaces or (pod.metadata.namespace in namespaces):
+            logging.debug("Getting Replicas for pod: %s in namespace: %s", pod.metadata.name, pod.metadata.namespace)
+            ownerRefs = pod.metadata.ownerReferences
+            namespace = pod.metadata.namespace
+
+            for ownerRef in ownerRefs:
+                if ownerRef.kind in supported_replica_objects and not not pod_replica_dict.get(ownerRef.name+namespace):
+
+                    logging.debug("Getting replica: %s, kind: %s, namespace: %s",
+                                  ownerRef.name, ownerRef.kind, namespace)
+
+                    if replicas_dict.get(ownerRef.name+namespace):
```

Comment:
the standard kubernetes notation for this would be directory or url style: `namespace + "/" + ownerRef.name`